var/spack/repos/builtin/packages/neuron/package.py

```diff
@@ -0,0 +1,240 @@
+    depends_on('pkg-config', type='build')
+    depends_on('python@2.6:', when='+python')
+    depends_on('mpi', when='+mpi')
+
+    # on bg-q mpi needs to be enabled : todo test this
+    conflicts('~mpi', when='platform=bgq')
+
+    # pgi compiler can't build static libraries
+    conflicts('%pgi', when='+static')
+
+    def url_for_version(self, version):
+        url = "http://www.neuron.yale.edu/ftp/neuron/versions/v{0}/nrn-{0}.tar.gz"
+        return url.format(version, version)
+
+    def patch(self):
+        # aclocal need complete include path especially on os x
+        pkgconf_inc = '-I %s/share/aclocal/' % (self.spec['pkg-config'].prefix)
+        libtool_inc = '-I %s/share/aclocal/' % (self.spec['libtool'].prefix)
+        newpath = 'aclocal -I m4 %s %s' % (pkgconf_inc, libtool_inc)
+        filter_file(r'aclocal -I m4', r'%s' % newpath, "build.sh")
+
+    def get_arch_options(self, spec):
+        options = []
+
+        if 'bgq' in self.spec.architecture:
+            options.extend(['--enable-bluegeneQ',
+                            '--host=powerpc64',
+                            '--without-memacs'])
+
+        if 'cray' in self.spec.architecture:
+            options.extend(['--without-memacs',
+                            '--without-nmodl',
+                            'cross_compiling=yes'])
+
+            # TODO: on cray systems we get an error while linking even if
+            # we use cc wrapper. Hence add explicitly add mpi library
+            if spec.satisfies('+mpi'):
+                options.append('LIBS=-lmpich')
+
+            # TODO: -pthread is not a valid pthread option for cray compiler.
+            # for now disable use of pthread with cray compiler.
+            if spec.satisfies('%cce'):
+                options.append('use_pthread=no')
+
+        return options
+
+    def get_opt_flags(self):
+        if 'bgq' in self.spec.architecture:
+            return '-O3 -qtune=qp -qarch=qp -q64 -qstrict -qnohot -g'
+        else:
+            return '-O2 -g'
+
+    def get_arch_dir(self):
+        if 'bgq' in self.spec.architecture:
+            arch = 'powerpc64'
+        elif 'cray' in self.spec.architecture:
+            arch = 'x86_64'
+        else:
+            arch = self.spec.architecture.target
+
+        return arch
+
+    def get_python_options(self, spec):
+        options = []
+
+        if spec.satisfies('+python'):
+            py_prefix = spec['python'].prefix
+            py_version = 'python{0}'.format(spec['python'].version.up_to(2))
+            python_exec = '%s/bin/%s' % (py_prefix, py_version)
+
+            options.append('--with-nrnpython=%s' % python_exec)
+            options.append('--disable-pysetup')
+
+            if spec.satisfies('+cross-compile'):
+                # on cross compile builds we have to provide
+                # PYINCDIR, PYLIB, PYLIBDIR etc
+
+                py_lib = spec['python'].prefix.lib
+                py_inc = '%s/include/%s' % (py_prefix, py_version)
```

Comment:
This looks like a bug in our definition of `LibraryList.names`. Since libraries can contain suffixes like `libfoo.dylib.1.0.6`, we search for the first period and strip everything after. This doesn't work when the library name contains a period. Let me see if I can fix it.